### PR TITLE
MAINT: fix incorrect TypeError --> ValueError

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

**Comment by bashtage** [comment]
Aside from needing to fix the test, these all look right to me.

**Comment by jbrockmendel** [comment]
Test fixed, CI fail is false-positive from coveralls.

**Comment by jbrockmendel** [comment]
@josef-pkt appveyor fail is in corrpsd, this is de facto green.

**Comment by jbrockmendel** [comment]
@josef-pkt green, i think this is the last of these

File: statsmodels/tsa/tests/test_tsa_tools.py [hunk 1]
```diff
@@ -653,7 +654,9 @@ def test_lagmat2ds_use_pandas(self):
 
     def test_3d_error(self):
         data = np.array(2)
-        assert_raises(TypeError, sm.tsa.lagmat2ds, data, 5)
+        with pytest.raises(ValueError):
+            sm.tsa.lagmat2ds(data, 5)
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply statsmodels/tsa/tests/test_tsa_tools.py]: The TypeError needed to be changed to ValueError.  While I was at it I changed it to use the modern context syntax.

[the remark this answers is not in the document]
Comment (josef-pkt) [reply statsmodels/tsa/tests/test_tsa_tools.py]: ok, this is not my "usual" neighborhood anymore, so ok with me

**Comment by jbrockmendel** [comment]
Well done sir.  I think #5192 is the last one in this genre at the moment

**Pull request merged**